core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableMergeJoin.java

```diff
@@ -222,6 +277,30 @@ public static boolean isMergeJoinSupported(JoinRelType joinType) {
     return mapping;
   }
 
+  private RelCollation extendCollation(RelCollation collation, List<Integer> keys) {
+    List<RelFieldCollation> fieldsForNewCollation = new ArrayList<>(keys.size());
+    fieldsForNewCollation.addAll(collation.getFieldCollations());
+    Set<Integer> keySet = new HashSet<>(keys);
+    for (RelFieldCollation rf : collation.getFieldCollations()) {
+      keySet.remove(rf.getFieldIndex());
+    }
+    for (Integer i : keySet) {
+      fieldsForNewCollation.add(new RelFieldCollation(i));
+    }
+    return RelCollations.of(fieldsForNewCollation);
+  }
+
+  private RelCollation removeCollationFieldsNotOnJoinKey(
```

Comment:
It is not obvious to get the meaning from method name. An example and comment will help.

Reply:
Let me add some comments with an example (a little bit complicated but let me try).

Reply:
I also think the method name is counter-intuitive. It is actually not removing anything, it just builds a new collation based on a "source" collation, considering only the fields which are part of the joinKeys. I would propose as name e.g. `reduceCollation`, `intersect`, or something like that..

Reply:
friendly reminder about these comments

Reply:
oops. I have missed this comment. 

Will address this one and another comment.

Reply:
`intersectCollecationAndJoinKey` is a reasonable name for me.